### PR TITLE
Explosions

### PR DESCRIPTION
**Pull request by ATLenormand**

### CODE REVIEW

File: Midstone/ComponentFramework/Scene5g.cpp [hunk 1]
```diff
@@ -230,12 +252,16 @@ void Scene5g::Update(const float deltaTime) {
 		frameCounter++;
 	}
 	shipPos.x -= deltaTime / 10;
-	shipModelMatrix = MMath::translate(shipPos) * MMath::rotate(90.0f, Vec3(0.0f, 1.0f, 0.0f));
+	shipModelMatrix = MMath::translate(shipPos * 40) * MMath::rotate(90.0f, Vec3(0.0f, 1.0f, 0.0f));
```
Comment (skywalker34) [inline Midstone/ComponentFramework/Scene5g.cpp:255]: What does 40 means?

File: Midstone/ComponentFramework/shaders/Explosion.glsl [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+} buf;
+
+layout(binding = 1) buffer layoutVel
+{
+    vec3 vel[];
+} buf2;
+
+
+float rand(float seed) { 
+    return fract(sin(seed) * 43758.5453123);
```
Comment (skywalker34) [inline Midstone/ComponentFramework/shaders/Explosion.glsl:22]: Whats this number?

File: Midstone/ComponentFramework/Explosion.cpp [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+bool Explosion::OnCreate(Camera* cam_, Shader* loadVerts, Mesh* mesh_, Model* model_)
+{
+	mesh = mesh_;
+	cam = cam_;
+
+	//create the buffer to hold the vertex positions between the vert stage and compute "stage"
+	glGenBuffers(1, &posBuffer);
+	glBindBuffer(GL_SHADER_STORAGE_BUFFER, posBuffer);
+	glBufferData(GL_SHADER_STORAGE_BUFFER, sizeof(Vec3) * 10000, NULL, GL_DYNAMIC_COPY); //null canse the loadVertsToBuffer shader will handle it
```
Comment (skywalker34) [inline Midstone/ComponentFramework/Explosion.cpp]: If the 10000 is same as line 20 maybe turn it into constant

File: Midstone/ComponentFramework/Scene5g.cpp (hunk 1 above)
Comment (ATLenormand) [reply Midstone/ComponentFramework/Scene5g.cpp:255]: Thats my test scene I just moved the test ship out of the way to test the explosion

File: Midstone/ComponentFramework/shaders/Explosion.glsl (hunk 2 above)
Comment (ATLenormand) [reply Midstone/ComponentFramework/shaders/Explosion.glsl:22]: just a large random float used for random number generation

File: Midstone/ComponentFramework/Explosion.cpp (hunk 3 above)
Comment (ATLenormand) [reply Midstone/ComponentFramework/Explosion.cpp]: yeah I should probably do that

**Review by skywalker34 — APPROVED** [review]

**Pull request merged**